iris-mpc-gpu/tests/e2e.rs

```diff
@@ -290,18 +293,26 @@ mod e2e_test {
                     matched_batch_request_ids,
                     anonymized_bucket_statistics_left,
                     anonymized_bucket_statistics_right,
+                    successful_reauths,
                     ..
                 } = res;
 
                 check_bucket_statistics(anonymized_bucket_statistics_left, num_devices)?;
                 check_bucket_statistics(anonymized_bucket_statistics_right, num_devices)?;
 
                 for (
-                    (((((req_id, &was_match), &idx), partial_left), partial_right), match_id),
+                    (
+                        (
+                            ((((req_id, &was_match), &was_reauth_success), &idx), partial_left),
+                            partial_right,
+                        ),
+                        match_id,
+                    ),
```

Comment:
we should probably be using `itertools::izip` for this, the large amount of tuple nesting is adding so much noise here.